scripts/dependency_intelligence.py

```
@@ -0,0 +1,341 @@
+
+def update_python_requirements(path: Path) -> None:
+    original = path.read_text(encoding="utf-8").splitlines()
+    output: list[str] = []
+    for raw_line in original:
+        stripped = raw_line.strip()
+        if not stripped or stripped.startswith("#") or stripped.startswith("-"):
+            output.append(raw_line)
+            continue
+        try:
+            requirement = Requirement(stripped)
+        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)